flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/generated/GeneratedWatermarkGeneratorSupplier.java

```diff
@@ -66,8 +71,10 @@ public GeneratedWatermarkGeneratorSupplier(
         } catch (Exception e) {
             throw new RuntimeException("Fail to instantiate generated watermark generator.", e);
         }
+
+        WatermarkEmitStrategy watermarkEmitStrategy = watermarkParams.getEmitStrategy();
```

Comment:
This class did not take into consideration that `watermarkParams` could be `null`.

Reply:
fixed